src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java

```diff
@@ -747,6 +784,7 @@ private static void register(Descriptor d) {
         d(KF, "DH",             P11DHKeyFactory,        s("DiffieHellman"),
                 m(CKM_DH_PKCS_KEY_PAIR_GEN, CKM_DH_PKCS_DERIVE));
         d(KF, "EC",             P11DHKeyFactory,
+                s("1.2.840.10045.2.1", "OID.1.2.840.10045.2.1"),    // Aliases for ECDSA and ECDH Public Key
```

Comment:
Please explain the need for this. Doesn't [KnownOIDs.EC](https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/src/java.base/share/classes/sun/security/util/KnownOIDs.java#L260) lead to `1.2.840.10045.2.1` being registered? If it is needed, why is it not also needed in newer Java versions?

Reply:
It seems these alias are also established in [`SunEC.putEntries()`](https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java#L296).

Reply:
I did not make this specific change; this was originally [a change made on z/OS](https://github.ibm.com/runtimes/openj9-openjdk-jdk11-zos/pull/1486#issuecomment-44725354) and brought over to LoZ.

As for why this is necessary, I noticed when testing that for some reason Java 11 uses `P11DHKeyFactory` for the `KeyFactory` service with the `EC` algorithm, where as Java 17 uses [`P11ECKeyFactory`](https://github.com/ibmruntimes/openj9-openjdk-jdk17/blob/e054b7448e3aea207e998a4eecc8717c0604dd14/src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java#L780) even though the `P11ECKeyFactory` class exists in both. 

`P11ECKeyFactory` uses the `SunEC` provider and like you noted, that provider does establish these aliases which is likely why I did not run into this same issue for Java 17. 

![image](https://user-images.githubusercontent.com/25011446/218566443-e918f369-fbc2-4cf0-8dd5-e33a7a6dfa69.png)

Reply:
I'm inclined to suggest we defer this addition until there's a clear explanation of why it's appropriate. As you note, in Java 17+, the code refers to `P11ECKeyFactory` instead of `P11DHKeyFactory`: perhaps that's the correct solution.

Reply:
I think that's a good idea. I'll remove this change from this PR and look into it further.